cpp/ql/src/semmle/code/cpp/ir/implementation/internal/TOperand.qll

```diff
@@ -165,12 +179,25 @@ module AliasedSSAOperands {
    * Returns the Phi operand with the specified parameters.
    */
   TPhiOperand phiOperand(
-    TAliasedSSAPhiInstruction useInstr, Aliased::Instruction defInstr,
+    Aliased::PhiInstruction useInstr, Aliased::Instruction defInstr,
     Aliased::IRBlock predecessorBlock, Overlap overlap
   ) {
     result = Internal::TAliasedPhiOperand(useInstr, defInstr, predecessorBlock, overlap)
   }
 
+  /**
+   * Returns the Phi operand with the specified parameters.
+   */
+  TPhiOperand reusedPhiOperand(
+    Aliased::PhiInstruction useInstr, Aliased::Instruction defInstr,
+    Aliased::IRBlock predecessorBlock, Overlap overlap
+  ) {
+    exists(Unaliased::IRBlock oldBlock |
+      predecessorBlock.getFirstInstruction() = oldBlock.getFirstInstruction() and
```

Comment:
If two blocks are fused together due to removal of unreachable code, is this still the right test? Maybe it should be comparing the last instruction of each predecessor block?